Signal/src/ViewControllers/AddToBlockListViewController.m

```diff
@@ -0,0 +1,346 @@
+                    callingCode:[NSString stringWithFormat:@"%@%@",
+                                 COUNTRY_CODE_PREFIX,
+                                 callingCode]
+                    countryCode:countryCode];
+}
+
+- (void)updateCountryWithName:(NSString *)countryName
+                  callingCode:(NSString *)callingCode
+                  countryCode:(NSString *)countryCode {
+
+    _callingCode = callingCode;
+
+    NSString *title = [NSString stringWithFormat:@"%@ (%@)",
+                       callingCode,
+                       countryCode.uppercaseString];
+    [_countryCodeButton setTitle:title
+                        forState:UIControlStateNormal];
+    [_countryCodeButton layoutSubviews];
+}
+
+- (void)setcallingCode:(NSString *)callingCode
+{
+    _callingCode = callingCode;
+
+    [self updateBlockButtonEnabling];
+}
+
+#pragma mark - Actions
+
+- (void)showCountryCodeView:(id)sender {
+    CountryCodeViewController *countryCodeController = [[UIStoryboard storyboardWithName:@"Registration" bundle:NULL]
+        instantiateViewControllerWithIdentifier:@"CountryCodeViewController"];
+    countryCodeController.delegate = self;
+    countryCodeController.shouldDismissWithoutSegue = YES;
+    UINavigationController *navigationController =
+        [[UINavigationController alloc] initWithRootViewController:countryCodeController];
+    [self presentViewController:navigationController animated:YES completion:[UIUtil modalCompletionBlock]];
+}
+
+- (void)blockButtonPressed:(id)sender
+{
+    [self tryToBlockPhoneNumber];
+}
+
+- (void)tryToBlockPhoneNumber
+{
+    if (![self hasValidPhoneNumber]) {
+        return;
+    }
+
+    NSString *possiblePhoneNumber = [self.callingCode stringByAppendingString:_phoneNumberTextField.text.digitsOnly];
+    PhoneNumber *parsedPhoneNumber = [PhoneNumber tryParsePhoneNumberFromUserSpecifiedText:possiblePhoneNumber];
+    OWSAssert(parsedPhoneNumber);
+
+    [_blockingManager addBlockedPhoneNumber:[parsedPhoneNumber toE164]];
+
+    UIAlertController *controller = [UIAlertController
+        alertControllerWithTitle:NSLocalizedString(@"BLOCK_LIST_VIEW_BLOCKED_ALERT_TITLE",
+                                     @"The title of the 'phone number blocked' alert in the block view.")
+                         message:[NSString
+                                     stringWithFormat:NSLocalizedString(@"BLOCK_LIST_VIEW_BLOCKED_ALERT_MESSAGE_FORMAT",
+                                                          @"The message format of the 'phone number blocked' alert in "
+                                                          @"the block view. It is populated with the blocked phone "
+                                                          @"number."),
+                                     [parsedPhoneNumber toE164]]
+                  preferredStyle:UIAlertControllerStyleAlert];
+
+    [controller addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"OK", nil)
```

Comment:
Agreed. I've already made this change in: https://github.com/WhisperSystems/Signal-iOS/pull/1927